### PR TITLE
chore: refactor handle error

### PR DESCRIPTION
**Pull request by Tuditi**
## Summary

Refactor error handling for usability with ledger devices.

...

## Changelog

```
- Added documentation on error handling
- Removed code duplication
- Updated base error handling
```

### Platforms

> Please select any platforms where your changes have been tested.

-   **Desktop**
    -   [ ] MacOS
    -   [ ] Linux
    -   [ ] Windows
-   **Mobile**
    -   [ ] iOS
    -   [ ] Android

### Instructions

> Please describe the specific instructions, configurations, and/or test cases necessary to **test and verify** that your changes work as intended.

...

## Checklist

> Please tick the following boxes that are relevant to your changes.

-   [x] I have followed the contribution guidelines for this project
-   [x] I have performed a self-review of my own code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have added or modified tests that prove my changes work as intended
-   [x] I have verified that new and existing unit tests pass locally with my changes
-   [x] I have verified that my latest changes pass CI workflows for testing and linting
-   [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/shared/lib/core/profile-manager/actions/events-handlers/handleNewOutputEvent.ts [hunk 1]
```diff
@@ -35,7 +35,7 @@ export async function handleNewOutputEventInternal(
         !get(allAccountActivities)[accountIndex].find((_activity) => _activity.id === output.outputId)
     const isNftOutput = output.output.type === OUTPUT_TYPE_NFT
 
-    if ((account.depositAddress === address && !output?.remainder) || isNewAliasOutput) {
+    if ((account?.depositAddress === address && !output?.remainder) || isNewAliasOutput) {
```
Comment (MarkNerdi) [inline packages/shared/lib/core/profile-manager/actions/events-handlers/handleNewOutputEvent.ts:38]: why is this required?

**Review by MarkNerdi — COMMENTED** [review]
Nice changes,   
1 small question

File: packages/shared/lib/core/profile-manager/actions/events-handlers/handleNewOutputEvent.ts (hunk 1 above)
Comment (Tuditi) [reply packages/shared/lib/core/profile-manager/actions/events-handlers/handleNewOutputEvent.ts:38]: An error was thrown when testing with ledger, so I added it for safety.

**Review by MarkNerdi — APPROVED** [review]
LGTM
